WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -233,7 +246,43 @@ final class AddEditCouponViewModel: ObservableObject {
         codeField = code
     }
 
-    func updateCoupon(coupon: Coupon, onUpdateFinished: @escaping () -> Void) {
+    func completeCouponAddEdit(coupon: Coupon, onUpdateFinished: @escaping () -> Void) {
+        if isCreationFlow {
+            createCoupon(coupon: coupon)
+        } else {
+            updateCoupon(coupon: coupon, onUpdateFinished: onUpdateFinished)
+        }
+    }
+
+    private func createCoupon(coupon: Coupon) {
+        ServiceLocator.analytics.track(.couponCreationInitiated)
```

Comment:
Since there's nothing getting updated here, but an entire coupon made from scratch instead, I thought that it would make sense to just trigger the couponCreationInitiated by itself, with no additional parameters.

Reply:
I totally missed the arguments detailed in your tracks P2 post, sorry 😅, it's fixed now!